src/hooks/privateChats.ts

```diff
@@ -215,3 +225,61 @@ export const useRecipientChat = (recipientId: string) => {
     recipient,
   };
 };
+
+export const usePrevious = <T>(value: T) => {
+  const ref = useRef<T>(value);
+  useEffect(() => {
+    ref.current = value;
+  });
+  return ref.current;
+};
+
+export const usePrivateChatNotification = (
+  privateChatMessages: WithId<PrivateChatMessage>[]
+) => {
+  const { user } = useUser();
+  const userId = user?.uid;
+
+  const { chatSettings } = useChatSidebarControls();
+
+  const [play] = useSound(newMessageSound, {
+    // `interrupt` ensures that if the sound starts again before it's
+    // ended, it will truncate it. Otherwise, the sound can overlap.
+    interrupt: true,
+  });
+
+  const privateChateMessagesSorted = useMemo(
+    () => Object.values(privateChatMessages).sort(chatSort),
+    [privateChatMessages]
+  );
+
+  const prevPrivateChatMessages = usePrevious(privateChateMessagesSorted[0]);
```

Comment:
Since you're only storing a single message here + better semantic naming:

```suggestion
  const previousNewestMessage = usePrevious(privateChateMessagesSorted[0]);
```

Reply:
After sorting with `chatSort`, is `privateChateMessagesSorted[0]` the most recent message, or the oldest message?

Reply:
I believe we only ever actually end up using the timestamp from this message, right? If so, we probably don't even need to save the whole message. We can probably just save the timestamp itself.

Something like:

```suggestion
  const lastMessageTimestampSec = usePrevious(privateChateMessagesSorted[0]?.ts_utc.seconds);
```